backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/scheduling/utils/VdsCpuUnitPinningHelper.java

```diff
@@ -55,8 +55,10 @@ public boolean isDedicatedCpuPinningPossibleAtHost(Map<Guid, List<VdsCpuUnit>> v
         int hostCoresPerSocket = host.getCpuCores() / host.getCpuSockets();
         int hostThreadsPerCore = host.getCpuThreads() / host.getCpuCores();
 
-        for (int socket = 0; socket < host.getCpuSockets(); socket++) {
-            if (hostCoresPerSocket * hostThreadsPerCore >= vcpus && getFreeCpusInSocket(cpuTopology, socket).size() < vcpus) {
+        for (int socket : getOnlineSockets(cpuTopology)) {
+            if (hostCoresPerSocket * hostThreadsPerCore == getCoresInSocket(cpuTopology, socket).size() &&
+                    getCoresInSocket(cpuTopology, socket).size() >= vcpus &&
```

Comment:
no need to execute `getCoresInSocket(cpuTopology, socket)` twice, right?
and what's the purpose of checking `hostCoresPerSocket * hostThreadsPerCore == getCoresInSocket(cpuTopology, socket).size()`? do you want to filter out all sockets that has an offline core?

Reply:
`hostCoresPerSocket * hostThreadsPerCore == getCoresInSocket(cpuTopology, socket).size()` is - if we have socket that fits the static host configuration (everything is online), then check if we can fit it into single socket.

Reply:
if this code is expected to change anyway, how about dropping this questionable part?

Reply:
this will break the unit tests, and so on. i think it's better to leave it as it is now and change everything on the upcoming PR.